### ISSUE TITLE
Roslyn should warn about null arguments to delegates' constructors

### ISSUE THREAD

**Issue by TessenR** [dotnet/roslyn#37984]
**Version Used**: 

```
Branch master (10 Aug 2019)
Latest commit c21548 by dotnet-automerge-bot:
Merge pull request #37896 from dotnet/merges/release/dev16.3-preview2-to-master

Merge release/dev16.3-preview2 to master
```

**Steps to Reproduce**:

Compile and run the following code:
```cs
#nullable enable
using System;

class C
{
  static void Main()
  {
    M(null);
  }
  static void M(Action? a)
  {
    var d = new D(a);
  }
}

delegate void D();
```

**Expected Behavior**:
Warning reported for `new D(a)` as delegates' constructors do not accept nulls.

**Actual Behavior**:
No warnings. The program crashes at runtime with 
```
System.ArgumentException: Delegate to an instance method cannot have null 'this'.
   at System.MulticastDelegate.CtorClosed(Object target, IntPtr methodPtr)
   at C.M(Action a)
   at C.Main()
```

**Comment by RikkiGibson** [comment]
The delegate constructor parameter doesn't seem to really have an expressible type. Intellisense shows it as 'void ()'. I don't know how it's represented internally in the compiler yet but it could be that we just have to special case this in nullable analysis.

**Comment by RikkiGibson** [comment]
The language specifies in §12.7.11.6 that a delegate constructor will throw NullReferenceException for a null argument, and users have no control over that. I would like to warn about it for any `new DelegateType(expr)` expression in a nullable enabled context, even if the delegate `DelegateType` was declared in a disabled annotations context.

**Comment by RikkiGibson** [comment]
Thanks for reporting this bug @TessenR.
